### PR TITLE
Feature/auth

### PR DESCRIPTION
**Pull request by Dilven**

### CODE REVIEW

File: src/Auth.tsx [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+import styled from 'react-emotion';
+import Button from 'components/Button'
+
+const StyledAuthForm = styled('form')`
+  box-shadow: 0 0 20px 4px #4384f5;
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+  width: 25%;
+  font-family:'Roboto';
```
Comment (kamilmateusiak) [inline src/Auth.tsx:11]: there is no 'Roboto' font in the app. If you want to use it you must include it ion index.html. Until now I was using Lato font

File: src/Auth.tsx [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+`
+const Label = styled('label')`
+  color:#999; 
+  font-size:18px;
+  font-weight:normal;
+  position:absolute;
+  pointer-events:none;
+  left:5px;
+  top:10px;
+  transition:0.2s ease all;
```
Comment (kamilmateusiak) [inline src/Auth.tsx:42]: It's not working as expected. OnBlur label is covering name filled in input ;/

File: src/Auth.tsx [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+  &:after {
+    right:50%; 
+  }
+`
+const Header = styled('h2')`
+  text-align:center; 
+  margin-bottom:30px; 
+`
+
+const AuthForm: React.SFC<{ singIn: (e: React.FormEvent<HTMLFormElement>) => void }> = ({ singIn }) => {
```
Comment (kamilmateusiak) [inline src/Auth.tsx:79]: it's better to always define an interface above component (more readable)

File: src/RequireAuth.tsx [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+export default function(ComposedComponent: React.ComponentType): React.ComponentType {
+  return class Authenticate extends Component<{}, IAuthenticateState> {
+    constructor(props: {}) {
+      super(props);
+      this.state = {
+        isLogged: false,
+      }
+    }
+
+    componentWillMount() {
```
Comment (kamilmateusiak) [inline src/RequireAuth.tsx:17]: don't use deprecated lifecycles: https://reactjs.org/docs/react-component.html#unsafe_componentwillmount

File: src/components/Chat/ChatContainer.tsx [hunk 5]
```diff
@@ -145,4 +146,4 @@ class App extends React.Component<{}, IState> {
   }
 }
 
-export default App;
+export default RequireAuth(App);
```
Comment (kamilmateusiak) [inline src/components/Chat/ChatContainer.tsx:149]: HOC name should be in camelcase with small first letter

**Review by kamilmateusiak — APPROVED** [review]

File: src/RequireAuth.tsx (hunk 4 above)
Comment (Dilven) [reply src/RequireAuth.tsx:17]: totally agree. "Copy paste" from old project :/

File: src/Auth.tsx (hunk 1 above)
Comment (Dilven) [reply src/Auth.tsx:11]: I forgot to include a font but if you want to use one font for the application, Lato is okay for me

**Pull request merged**